### PR TITLE
Support inherited annotations for all major objects

### PR DESCRIPTION
**Pull request by FxKu**
In preparation for an inherited annotations feature, similar to inherited labels, to estimate how much it would take.

### CODE REVIEW

File: pkg/cluster/connection_pooler.go [hunk 1]
```diff
@@ -334,12 +334,15 @@ func (c *Cluster) generateConnectionPoolerDeployment(connectionPooler *Connectio
 		return nil, err
 	}
 
+	annotations := c.annotationsSet(nil)
+	annotations = c.AnnotationsToPropagate(annotations)
+
```
Comment (RafiaSabih) [inline pkg/cluster/connection_pooler.go:339]: Why not directly passing nil to AnnotationsToPropagate?

Comment (RafiaSabih) [inline pkg/cluster/connection_pooler.go:339]: Why not, annotations := c.AnnotationsToPropagate(c.annotationsSet(nil))

File: pkg/cluster/connection_pooler.go [hunk 2]
```diff
@@ -866,8 +869,9 @@ func (c *Cluster) syncConnectionPoolerWorker(oldSpec, newSpec *acidv1.Postgresql
 		}
 	}
 
-	newAnnotations := c.AnnotationsToPropagate(c.ConnectionPooler[role].Deployment.Annotations)
-	if newAnnotations != nil {
+	newAnnotations := c.annotationsSet(c.ConnectionPooler[role].Deployment.Annotations)
+	newAnnotations = c.AnnotationsToPropagate(newAnnotations)
```
Comment (RafiaSabih) [inline pkg/cluster/connection_pooler.go:873]: Again, could be just one line like the previous comment.

File: pkg/cluster/util.go [hunk 3]
```diff
@@ -271,6 +271,33 @@ func (c *Cluster) getTeamMembers(teamID string) ([]string, error) {
...
+		}
+	}
+
+	if len(annotations) > 0 {
+		return annotations
+	}
+
+	return nil
+}
+
```
Comment (RafiaSabih) [inline pkg/cluster/util.go:300]: Not sure if returning nil is used anywhere, can simply return annotations.

Comment (FxKu) [reply pkg/cluster/util.go:300]: it should return nil if there are no Annotations defined in the config and on the resource. For example `TestGeneratePodDisruptionBudget` fails for me if I return an empty map as this test does not define any Annotations at all. I have also added this behavior to the `AnnotationsToPropagate` function, too.

**Comment by FxKu** [comment]
:+1:

**Comment by RafiaSabih** [comment]
:+1:

**Pull request merged**